backend/api/v1/routers/datasets.py

```
@@ -0,0 +1,269 @@
+    except Exception as e:
+        logger.error(f"Unexpected error creating dataset: {e}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Internal server error"
+        )
+
+
+@router.get("/{name}", response_model=DatasetResponse)
+async def get_dataset(
+    name: str,
+    service: DatasetManagementService = Depends(get_dataset_service)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)